common: tools: add string compare function

mlohvynenko opened the pull request:
This patch introduces a string compare function that allows comparison of two strings, supporting both
case-sensitive and case-insensitive modes.

mlohvynenko committed:
common: tools: add string compare function

This patch introduces a string compare function that
allows comparison of two strings, supporting both
case-sensitive and case-insensitive modes.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mykola-kobets-epam reviewed (approved):
`Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>`

al1img reviewed (approved):
`Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>`

mlohvynenko committed:
iam: nodeinfoprovider: use string case compare method

This patch replaces the custom case-insensitive string comparison
function with String::Compare method.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
sm: launcher: use case-insensitive string comparison

Use case-insensitive string comparison instead of converting
string to lower case for OS check.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mykola-kobets-epam reviewed (approved):
`Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>`

MykolaSuperman reviewed (approved):
`Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>`
